benches/micro.rs

```diff
@@ -0,0 +1,130 @@
+#![allow(missing_docs)]
+
+use criterion::{criterion_group, criterion_main, Criterion, Throughput};
+
+use fsst::{CompressorBuilder, Symbol};
+
+fn one_megabyte(seed: &[u8]) -> Vec<u8> {
+    seed.iter().copied().cycle().take(1024 * 1024).collect()
+}
+
+fn bench_compress(c: &mut Criterion) {
+    let mut group = c.benchmark_group("compress-overhead");
+    group.bench_function("compress-word", |b| {
+        let mut compressor = CompressorBuilder::new();
+        compressor.insert(Symbol::from_u8(b'a'), 1);
+        let compressor = compressor.build();
+
+        let mut output = [0u8, 0u8];
+
+        b.iter(|| unsafe {
+            compressor.compress_word('a' as u64, output.as_mut_ptr());
+        });
+    });
+
+    // Reusable memory to hold outputs
+    let mut output_buf: Vec<u8> = Vec::with_capacity(1_024 * 1024 * 2);
+
+    group.bench_function("compress-hashtab", |b| {
+        // We create a symbol table and an input that will execute exactly one iteration,
+        // in the fast compress_word pathway.
+        let mut compressor = CompressorBuilder::new();
+        compressor.insert(Symbol::from_slice(b"abcdefgh"), 8);
+        let compressor = compressor.build();
+
+        b.iter(|| unsafe {
+            compressor.compress_into(
+                b"abcdefghabcdefghabcdefghabcdefghabcdefghabcdefghabcdefghabcdefgh",
+                &mut output_buf,
+            );
+        });
+    });
+
+    group.bench_function("compress-twobytes", |b| {
+        // We create a symbol table and an input that will execute exactly one iteration,
+        // in the fast compress_word pathway.
+        let mut compressor = CompressorBuilder::new();
+        compressor.insert(Symbol::from_slice(&[b'a', b'b', 0, 0, 0, 0, 0, 0]), 8);
+        let compressor = compressor.build();
+
+        b.iter(|| unsafe {
+            compressor.compress_into(b"abababababababab", &mut output_buf);
+        });
+    });
+    group.finish();
```

Comment:
These test the relative speeds of the two compression pathways: one where we're able to skip the hashtable lookup and one where we are not. Skipping the lookup is nearly twice as fast

![image](https://github.com/user-attachments/assets/851c8e41-0ac0-4f78-bfff-e90d43e52ab1)